installer/windows/Product.wxs

```diff
@@ -57,9 +57,18 @@
           Account="[SERVICEACCOUNT]"
           Password="[SERVICEPASSWORD]"
           ErrorControl="normal"
-          Arguments="run --config-dir &quot;[INSTALLFOLDER]\&quot; --identity-dir &quot;[IDENTITYDIR]\&quot; --binary-location &quot;[INSTALLFOLDER]\storagenode.exe&quot; --log &quot;[INSTALLFOLDER]\storagenode-updater.log&quot;"
-          />
-        <ServiceControl Id="StoragenodeUpdaterStartService" Start="install" Stop="both" Remove="uninstall" Name="storagenode-updater" Wait="yes" />
+          Arguments="run --config-dir &quot;[INSTALLFOLDER]\&quot; --identity-dir &quot;[IDENTITYDIR]\&quot; --binary-location &quot;[INSTALLFOLDER]\storagenode.exe&quot; --log &quot;[INSTALLFOLDER]\storagenode-updater.log&quot;">
+            <util:ServiceConfig
+              ServiceName="storagenode-updater"
+              ResetPeriodInDays="1"
+              RestartServiceDelayInSeconds="60"
+              FirstFailureActionType="restart"
+              SecondFailureActionType="runCommand"
+              ThirdFailureActionType="runCommand"
+              ProgramCommandLine="cmd /c &quot;(echo %date% %time% recovery triggered to execute in 30 minutes &amp;&amp; timeout 1800 >NUL &amp;&amp; echo %date% %time% recovery command executed: &amp;&amp; copy /y &quot;[INSTALLFOLDER]storagenode-updater.old.exe&quot; &quot;[INSTALLFOLDER]storagenode-updater.exe&quot; &amp;&amp; net start storagenode-updater) >> &quot;[INSTALLFOLDER]storagenode-updater.log&quot;&quot;"
```

Comment:
@kaloyan-raev did you test this part `echo %date% %time% recovery triggered to execute in 30 minutes`? For me it looks that it will be printed after service restart (at the end)

Reply:
Yes, I tested it. It is printed before the timeout is executed.

The only issue with this big command line is that both `echo` statements print the same date and time. This is because these variables are evaluated at the beginning of the execution of the whole big line, not when the echo happens.

I am not sure how to resolve it, but it is a minor issue.

Reply:
Cool, thanks! I agree that the same date and time is not very important